src/l500/l500-depth.cpp

```diff
@@ -344,14 +324,21 @@ namespace librealsense
             auto dp = std::find_if(requests.begin(), requests.end(), [](std::shared_ptr<stream_profile_interface> sp)
             {return sp->get_stream_type() == RS2_STREAM_DEPTH;});
 
-            const int XGA = 0;
-            const int VGA = 1;
-
-            if (dp != requests.end() && supports_option(RS2_OPTION_CAMERA_MODE))
+            if (dp != requests.end() && supports_option(RS2_OPTION_SENSOR_MODE))
             {
-                auto&& camera_mode_option = get_option(RS2_OPTION_CAMERA_MODE);
+                auto&& sensor_mode_option = get_option(RS2_OPTION_SENSOR_MODE);
                 auto vs = dynamic_cast<video_stream_profile*>((*dp).get());
-                camera_mode_option.set(vs->get_width() == 640 || vs->get_height() == 640 ? VGA : XGA);
+                if (supports_option(RS2_OPTION_VISUAL_PRESET))
+                {
+                    auto&& preset_option = get_option(RS2_OPTION_VISUAL_PRESET);
+                    if (preset_option.query() == RS2_L500_VISUAL_PRESET_CUSTOM)
+                    {
+                        if(sensor_mode_option.query() != get_resolution_from_width_height(vs->get_width(), vs->get_height()))
+                            throw  std::runtime_error(to_string() << "sensor mode option is incompatible with requsted resolution");
```

Comment:
Please specify the requested resolution and the configured sensor mode, and make the comment instructive rather than descriptive: e.g
"The sensor mode ..(X/V) GA.. must conform to the selected resolution [<w>,<h>]"